lib/rules/require-form-method.js

```diff
@@ -0,0 +1,90 @@
+'use strict';
+
+const Rule = require('./base');
+const createErrorMessage = require('../helpers/create-error-message');
+
+const ERROR_MESSAGE = 'All `<form>` elements should have `method` attribute with value of `%`';
+
+// Form `method` attribute keywords:
+// https://html.spec.whatwg.org/multipage/form-control-infrastructure.html#attr-fs-method
+const DEFAULT_CONFIG = {
+  allowedMethods: ['POST', 'GET'],
+};
+
+module.exports = class RequireFormMethod extends Rule {
+  logNode({ node, message }) {
+    return this.log({
+      message,
+      line: node.loc && node.loc.start.line,
+      column: node.loc && node.loc.start.column,
+      source: this.sourceForNode(node),
+      isFixable: false,
+    });
+  }
+
+  parseConfig(config) {
+    let configType = typeof config;
+
+    switch (configType) {
+      case 'boolean':
+        return config ? DEFAULT_CONFIG : false;
+      case 'object':
+        if (Array.isArray(config.allowedMethods)) {
+          return {
+            allowedMethods: config.allowedMethods.map((m) => String(m).toUpperCase()),
```

Comment:
Can you throw an exception if any unexpected methods are passed in?

Reply:
Added. I also added "dialog" method type as by HTML spec to defaults list.